dev-client/src/model/soilId/soilIdRanking.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {
+  DataBasedSoilMatch,
+  LocationBasedSoilMatch,
+} from 'terraso-client-shared/graphqlSchema/graphql';
+import {SoilIdResults} from 'terraso-client-shared/soilId/soilIdTypes';
+
+export const getTopMatch = (
+  results: SoilIdResults,
+): LocationBasedSoilMatch | DataBasedSoilMatch | undefined => {
+  const locationBased = results.locationBasedMatches;
+  const dataBased = results.dataBasedMatches;
+
+  if (dataBased.length > 0) {
+    return dataBased.reduce(
+      (prev, current) =>
+        current.combinedMatch.rank > prev.combinedMatch.rank ? current : prev,
+      dataBased[0],
```

Comment:
Issue: This is backwards. The item with the smallest rank should be returned (it might always be the item with rank 0 but I don't know for usre)

Reply:
Good catch!